tezos/interop/src/ffi.rs

```diff
@@ -403,3 +420,79 @@ pub fn assert_encoding_for_protocol_data(
         })
     })
 }
+
+pub fn apply_block_result_metadata(
```

Comment:
@bkontur this function and the one immediately after are the ones that should be called to produce the JSON